File: javascript/6_kyu/iq_test.js
// IQ Test
// https://www.codewars.com/kata/552c028c030765286c00007d/train/javascript

function iqTest(numbers) {
  numbers = numbers.split(" ").map(function(a) {
    return parseInt(a, 10);
  })

  var split_numbers = {"evens": [], "odds": []}

  for (var i = 0; i < numbers.length; i++) {
    if ( numbers[i] % 2 == 0 ) {
      split_numbers["evens"].push(numbers[i]);
    } else {
      split_numbers["odds"].push(numbers[i]);
    }
  }

  var search;

  if (split_numbers["evens"].length == 1) {
    search = split_numbers["evens"][0];
  } else {
    search = split_numbers["odds"][0];
  }

  for (var i = 0; i < numbers.length; i++) {
    if (numbers[i] == search) {
      return i + 1;
    }
  }
}

console.log(iqTest("2 4 7 8 10"));
console.log(iqTest("1 2 1 1"))

File: python/6_kyu/iq_test.py
# IQ Test
# https://www.codewars.com/kata/552c028c030765286c00007d/train/python

# Solution 1: Map Function
def iq_test(numbers):
  nums =  list(map(int, numbers.split(" ")))
  even_or_odd = {"even": [], "odd": []}
  
  for i in nums:
    if i % 2 == 0:
      even_or_odd['even'].append(i)
    else:
      even_or_odd['odd'].append(i)
  
  for key in even_or_odd:
    if len(even_or_odd[key]) == 1:
      for i in range(0, len(nums)):
        if nums[i] == even_or_odd[key][0]:
          return i + 1

# Solution 2: Index, Pythonic Solution
def iq_test(numbers):
  num = [int(i) % 2 for i in numbers.split(" ")]
  if num.count(0) > 1:
    return num.index(1) + 1
  else:
    return num.index(0) + 1

print(iq_test("2 4 7 8 10"))
print(iq_test("1 2 1 1"))
print(iq_test("1 2 2"))